specifications/secure-launch-specification.rst

```diff
@@ -0,0 +1,719 @@
+
+.. include:: <isonum.txt>
+
+###########################
+Secure Launch Specification
+###########################
+
+.. class:: center
+
+**Version:** 1.0.0  |middot|  **Authors:** TrenchBoot Team
+
+.. sectnum::
+
+.. contents:: Table of Contents
+
+Introduction
+============
+
+Like most hardware capabilities implemented by different manufactures, each
+platform has its own unique way of implementing a capability. Dynamic Launch is
+no different in this manner. The TrenchBoot project is striving to provide a
+unified approach to using Dynamic Launch across different platforms.  One
+aspect of Dynamic Launch that varies across platforms is the launch related
+data. For example, on Intel there is a series of Intel defined configuration
+data structures that must be present, as well as a user-defined data structure.
+Whereas on AMD, only the layout of an AMD Secure Loader Block is defined and
+thus, any configuration structures are SLB implementation specific.
+
+To provide a unified experience in passing configuration and other meta-data to
+a TrenchBoot Secure Launch entry point for a kernel, this specification
+provides the platform-agnostic Secure Launch Resource Table (SLRT) along with
+details on how it will be implemented for each platform supported.
+
+Terminology
+===========
+
+Requirements Language
+---------------------
+
+The key words **"MUST"**, **"MUST NOT"**, **"REQUIRED"**, **"SHALL"**,
+**"SHALL NOT"**, **"SHOULD"**, **"SHOULD NOT"**, **"RECOMMENDED"**,  **"MAY"**,
+and **"OPTIONAL"** in this document are to be interpreted as described in
+:rfc:`2119`.
+
+
+Acronyms
+--------
+
+:DCE: Dynamic Configuration Environment (*eg. Intel ACM/AMD SLB*)
+:DLE: Dynamic Launch Event (*eg. Intel GETSEC[SENTER]/AMD SKINIT*)
+:DLME: Dynamic Launch Measured Environment (*eg. Operating System/Hypervisor*)
+:DRTM: Dynamic Root of Trust Measurement
+
+
+Secure Launch Architecture
+==========================
+
+
+Secure Launch aware Bootloader
+------------------------------
+
+A bootloader is Secure Launch aware if it understands how to load or provide a
+DLE Handler and configure an SLRT.
+
+Bootloader Types
+~~~~~~~~~~~~~~~~
+
+A Secure Launch aware bootloader can be categorized as one of three types,
+
+:monolithic: This is a bootloader that contains the DLE Handler.
+:initializing: This is a bootloader that loads an external DLE Handler and
+               executes a supplemental bootloader.
+:supplemental: This is a bootloader that is capable of calling a DLE Handler.
+
+Dynamic Launch Event Handler
+----------------------------
+
+The DLE (Dynamic Launch Event) Handler consumes the SLRT and invokes the
+platform's dynamic launch event. This specification seeks to standardize the
```

Comment:
Sometimes we use the small letter for dynamic launches and sometimes for big ones; why? We should try to be more consistent.

Reply:
When used as part of a name, it should be capitalized. When it is referring to the general concept it should be lower cased. With that said, the instance you point out should not have even been spelled out, the acronym should have been used.

fixed.